packages/internal/src/types.ts

```diff
@@ -74,6 +74,26 @@ export declare class SyncCallback<
  */
 export type IsPrimitive<T> = [T] extends [Primitive] ? true : false;
 
+export type IsStrictAny<T> = 0 extends 1 & T ? false : true;
+export type IsUnknown<T> = unknown extends T ? IsStrictAny<T> : false;
+export type IsRemotable<T> = T extends RemotableObject
```

Comment:
this really shouldn't be necessary. 
```suggestion
// XXX https://github.com/endojs/endo/issues/2979
export type IsRemotable<T> = T extends RemotableObject
```

Reply:
Gotcha.